apps/server/src/modules/oauth/uc/oauth.uc.ts

```diff
@@ -0,0 +1,137 @@
+import { Injectable } from '@nestjs/common';
+import { ILogger, Logger } from '@src/core/logger';
+import axios, { AxiosResponse } from 'axios';
+import jwtDecode from 'jwt-decode';
+import { SystemRepo } from '@shared/repo/system';
+import { UserRepo } from '@shared/repo';
+import { System, User } from '@shared/domain';
+import { FeathersJwtProvider } from '@src/modules/authorization/feathers-jwt.provider';
+import { Configuration } from '@hpi-schul-cloud/commons';
+import { Payload } from '../controller/dto/payload';
+import { OauthTokenResponse } from '../controller/dto/oauthTokenResponse';
+import { AuthorizationQuery } from '../controller/dto/authorization.query';
+import { OAuthResponse } from '../controller/dto/oauth.response';
+
+@Injectable()
+export class OauthUc {
+	private logger: ILogger;
+
+	constructor(
+		private readonly systemRepo: SystemRepo,
+		private readonly userRepo: UserRepo,
+		private readonly jwtService: FeathersJwtProvider
+	) {
+		this.logger = new Logger(OauthUc.name);
+	}
+
+	// 0- start Oauth Process
+	async startOauth(query: AuthorizationQuery, systemId: string): Promise<OAuthResponse> {
+		try {
+			// get the authorization code
+			const code: string = this.extractCode(query);
+			// get the Tokens using the authorization token
+			const queryToken: OauthTokenResponse = await this.requestToken(code, systemId);
+			// extract the uuid from the token
+			const uuid = await this.decodeToken(queryToken.id_token);
+			// get the user using the uuid
+			const user: User = await this.findUserById(uuid);
+			// create JWT for the user
+			const jwt: string = await this.getJWTForUser(user);
+			// send response back
+			const HOST = Configuration.get('HOST') as string;
+			const redirectUri = HOST.concat('/dashboard');
+			const response: OAuthResponse = new OAuthResponse({
+				jwt,
+				redirectUri,
+			});
+			return response;
+		} catch (error) {
+			this.logger.log(error);
+		}
+		// send error response back
+		const HOST = Configuration.get('HOST') as string;
+		const redirectUri = HOST.concat('/login');
+		const response: OAuthResponse = new OAuthResponse({
+			redirectUri: `${redirectUri}?error=OauthLoginFailed`,
+		});
+		return response;
+	}
+
+	/**
+	 * @query query input that has either a code or an error
+	 * @return authorization code or throws an error
+	 */
+	extractCode(query: AuthorizationQuery): string {
+		if (query.code) return query.code;
+		if (query.error) throw new Error(query.error);
+		throw new Error('Authorization Query Object has no authorization code or error');
```

Comment:
this method is validation,just check if(query.code) and return it otherwise throw. name it checkAuthorizationCode()...

Reply:
Authorization Query has a code (authorization code) or an error (authorization error). The iServ server (oauth server) sends either a code or an error message, according to the error that has happened. In the next release, some of these error messages should be sent to the front end. 
In all our tests our Get endpoint was called either with a query of a code or an error. So theoretically the line 67 will not be called, just in a case that we haven't seen, and that is not in any documentation found.

Reply:
Method name was changed to checkAuthorizationCode as requested.